### ISSUE TITLE
Can I connect to my ps4 directly with IP address?

### ISSUE THREAD

**Issue by Zhendong404** [MysteryDash/Offline-PS4-Remote-Play#19]
I'm in a special situation, my PS4 and PC are on the same LAN, but different network segment(e.g. 10.13.71.0/24 and 10.110.41.0/24). The Remote Play can't find my PS4 on LAN, so it always connect through Internet, but my Internet bandwidth is not enough for this.
I'm thinking of patching the remote play so it can connect my ps4 directly with an IP. Any suggestions?

**Comment by MysteryDash** [comment]
Maybe adding a rule in your hosts file might work.  
For example, adding (source then destination) 10.13.71.x 10.110.41.y with x any number that isn't currently used and y your PS4's actual IP address might trick the Remote Play into believing that your PS4 is on your local network, I haven't tested that but it's worth a try.

**Comment by Zhendong404** [comment]
I finally find out that remote play use broadcasting to search PS4, so it's not that simple. Now I'm trying with VPN. Steam In-home streaming works with this way, still some more works to do with PS4.

**Comment by MysteryDash** [comment]
Okay, let me know if it works!

**Comment by Zhendong404** [comment]
I just made it!
Here's what I have done, hope this could be useful.
I started a VM on another PC, which is under the same router with my PS4, and installed CentOS on it.
Then I was basically following this tutorial with some modifications:
[https://stewartadam.io/blog/2016/04/13/using-openvpn-connection-play-games-while-abroad-using-streams-home-streaming](https://stewartadam.io/blog/2016/04/13/using-openvpn-connection-play-games-while-abroad-using-streams-home-streaming)
1. add <push "0.0.0.0 0.0.0.0 {server ip}"> into bridged.conf
2. replace the content of up.sh with
    /sbin/brctl addif br0 tap0
    /sbin/ifconfig tap0 0.0.0.0 promisc up 
3. add a user to group "openvpn_pw"

On my client PC, after connected to the server, Steam In-home Streaming worked, but remote play still couldn't find my PS4 on LAN. After remote play connected to PS4 through Internet, I restarted remote play, and this time it found PS4 on LAN.
